tests/backend/test_twomode_squeezing_operation.py

```diff
@@ -74,3 +74,28 @@ def test_two_mode_squeezing(self, setup_backend, r, p, cutoff, pure, tol):
                     tmsv2 = get_amplitude(k, r, p) * np.conj(get_amplitude(l, r, p))
 
                     assert np.allclose(state.data[t], tmsv2, atol=tol, rtol=0)
+
+
+    @pytest.mark.parametrize("r", MAG)
+    @pytest.mark.parametrize("p", PHASE)
+    def test_squeezing_on_mode_subset(self, setup_backend, r, p, cutoff, pure, tol):
+        r""" Test two-mode squeezing on vacuum-state for both pure states and
+        mixed states on different mode subsets with the amplitude given by
+	        :math:`\delta_{kl} \frac{e^{in\phi} \tanh^n{r}}{\cosh{r}}`
+        """
+        z = r * np.exp(1j * p)
+        n_modes = 4
+        for modes in it.combinations(range(n_modes), 2):
+            backend = setup_backend(n_modes)
+            backend.two_mode_squeeze(z, *modes)
+
+            state = backend.state()
+
+            state_data = state.reduced_dm(list(modes))
+
+            for k in it.product(range(cutoff), repeat=2):
+                for l in it.product(range(cutoff), repeat=2):
```

Comment:
Same comment here re: parametrize 🙂 Note that 'stacking' decorators is equivalent to nesting for loops.

Reply:
One downside here though: this would require performing the `two_mode_squeeze` method many more times